src/core/server/saved_objects/saved_objects_service.test.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+jest.mock('./migrations/kibana/kibana_migrator');
+
+import { SavedObjectsService, SavedObjectsSetupDeps } from './saved_objects_service';
+import { CoreContext } from '../core_context';
+import { mockCoreContext } from '../core_context.mock';
+import { KibanaMigrator } from './migrations/kibana/kibana_migrator';
+import { of } from 'rxjs';
+import elasticsearch from 'elasticsearch';
+
+let coreContext: CoreContext;
+beforeEach(() => {
+  coreContext = mockCoreContext.create();
+});
+
+afterEach(() => {
+  jest.clearAllMocks();
+});
+
+describe('SavedObjectsService', () => {
+  describe('#setup()', () => {
+    it('creates a KibanaMigrator which retries NoConnections errors from callAsInternalUser', async () => {
+      let i = 0;
+      const clusterClient = {
+        callAsInternalUser: jest
+          .fn()
+          .mockImplementation(() =>
+            i++ <= 2
+              ? Promise.reject(new elasticsearch.errors.NoConnections())
+              : Promise.resolve('success')
+          ),
+      };
+
+      const soService = new SavedObjectsService(coreContext);
+      const coreSetup = ({
+        elasticsearch: { adminClient$: of(clusterClient) },
+        legacy: { uiExports: {}, pluginExtendedConfig: {} },
+      } as unknown) as SavedObjectsSetupDeps;
+
+      await soService.setup(coreSetup);
+
+      return expect((KibanaMigrator as jest.Mock).mock.calls[0][0].callCluster()).resolves.toMatch(
```

Comment:
I want to be sure that `SavedObjectsService.setup()` creates a `KibanaMigrator` which will retry any callCluster calls. Are you suggesting that I move the `retryCallCluster` into the `KibanaMigrator`? I think this will probably be a bit simpler to test, so I'll give that a try.